pkg/controller/controller.go

```diff
@@ -362,6 +369,24 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 		return nil, fmt.Errorf("claim Selector is not supported")
 	}
 
+	createVolumeRequestParameters := options.Parameters
+	performInTreeTranslation := false
+	if p.handlesMigrationFromInTreePlugin {
+		storageClassName := options.PVC.Spec.StorageClassName
+		storageClass, err := p.client.StorageV1().StorageClasses().Get(*storageClassName, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get storage class named %s: %v", *storageClassName, err)
+		}
+		if storageClass.Provisioner == p.handlesMigrationFromInTreePluginName {
+			klog.V(2).Infof("Perform CSI migration for intree plugin %s", storageClass.Provisioner)
```

Comment:
be more specific here. `CSI Migration: translating storage class parameters for in-tree plugin %s`?